### PR TITLE
Let SQS AddPermission call actually influence policy attribute

### PR DESCRIPTION
**Pull request by dfangl**
## Motivation
Currently, SQS AddPermission and RemovePermission calls do only save the permissions in some internal datastructure - they are never returned with sqs.GetQueueAttributes. This is problematic, since this is a necessity for cross account evaluation.

You could just use SetQueueAttributes on the `Policy` attribute but we should match AWS behavior.

## Changes
* Rework SQS permission management, by adapting the stored policy
* Add AWS validated test for permission lifecycles.

### CODE REVIEW

**Comment by dfangl** [comment]
Tests against pro will be fixed with #8557

File: localstack/services/sqs/models.py [hunk 1]
```diff
@@ -163,13 +163,6 @@ def __repr__(self):
         return f"SqsMessage(id={self.message_id},group={self.message_group_id})"
 
 
-class Permission(NamedTuple):
-    # TODO: just a placeholder for real policies
```
Comment (thrau) [inline localstack/services/sqs/models.py:167]: nice to finally get that todo done :100:

File: localstack/services/sqs/models.py [hunk 2]
```diff
@@ -496,6 +488,74 @@ def validate_queue_attributes(self, attributes):
             if k not in valid:
                 raise InvalidAttributeName(f"Unknown Attribute {k}.")
 
+    def add_permission(self, actions: list[str], account_ids: list[str], label: str) -> None:
```
Comment (thrau) [inline localstack/services/sqs/models.py:491]: super minor nit: i think having label as the first argument would be a more consistent API

**Review by thrau — APPROVED** [review]
excellent! :rocket:

**Pull request merged**